packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -231,14 +233,30 @@ class Icon extends StatelessWidget {
   /// specified, either directly using this property or using [Directionality].
   final TextDirection? textDirection;
 
+  /// Whether the size is affected by the context's text scaler.
+  ///
+  /// If true, the icon size is going to be scaled by the [TextScaler] of the
+  /// context's [MediaQuery].
+  ///
+  /// Defaults to the nearest [IconTheme]'s
+  /// [IconThemeData.considerTextScale].
+  final bool? considerTextScale;
+
   @override
   Widget build(BuildContext context) {
     assert(this.textDirection != null || debugCheckHasDirectionality(context));
     final TextDirection textDirection = this.textDirection ?? Directionality.of(context);
 
     final IconThemeData iconTheme = IconTheme.of(context);
 
-    final double? iconSize = size ?? iconTheme.size;
+    final bool? considerTextScale = this.considerTextScale ?? iconTheme.considerTextScale;
+
+    final double? tentativeIconSize = size ?? iconTheme.size;
```

Comment:
This should default to 14 when null.

Reply:
Out of curiosity, why?

I can't see anything in the current code that makes it default to `14.0`, so this is essentially a new behavior, isn't it?

Reply:
It's the default font size the text layout libraries use when the font size isn't specified (it currently happens in flutter/engine so you probably won't find that logic in flutter/flutter).

Reply:
If the engine handles it whey should we specify it now?

Reply:
because we are using the font size to scale the icon right? It doesn't make sense to skip scaling because there's no explicitly specified font size.

Reply:
Hmm, I get what you are saying now... In this case, do we have any way to knowing this value instead of hardcoding it?

I mean, if the engine ever changes its "internal value" from `14.0` to anything else, it's not hard to forget to update it here, so we could potentially and silently introduce a regression here...

Reply:
Yeah good point. There's a private constant defined in `text_painter.dart` and another one in `widget_span.dart`. You could make the text_painter one public and remove the other.

Reply:
Done.